src/Parsing.hs

```diff
@@ -223,7 +224,8 @@ symbolSegment :: Parsec.Parsec String ParseState String
 symbolSegment = do sym <- Parsec.many1 validInSymbol
                    incColumn (length sym)
                    return sym
-  where validInSymbol = Parsec.choice [Parsec.letter, Parsec.digit, Parsec.oneOf validCharacters]
+  where validInSymbol = Parsec.choice [Parsec.letter, Parsec.digit, Parsec.oneOf validCharacters, highCharacters]
+        highCharacters = Parsec.satisfy ((> 127) . ord)
```

Comment:
You probably shouldn’t listen to me tonight, because I clearly am not on top of my game, but I suppose this will also allow Unicode whitespace?

Reply:
Fun fact: we allow this now(I tried it with an Em Quad), but Clang does not:
```
> (def   1)
>  
  : Int
Defined at line 1, column 1 in 'REPL'
> (defn main []  )
> :bx
/Users/veitheller/.carp/out/main.c:1424:5: warning: treating Unicode character as whitespace
      [-Wunicode-whitespace]
int  ;
    ^
/Users/veitheller/.carp/out/main.c:1424:1: warning: declaration does not declare anything
      [-Wmissing-declarations]
int  ;
^~~
/Users/veitheller/.carp/out/main.c:2903:9: warning: treating Unicode character as whitespace
      [-Wunicode-whitespace]
          = 1;
        ^
/Users/veitheller/.carp/out/main.c:2903:13: error: expected expression
          = 1;
          ^
/Users/veitheller/.carp/out/main.c:8758:12: warning: treating Unicode character as whitespace
      [-Wunicode-whitespace]
    return  ;
           ^
/Users/veitheller/.carp/out/main.c:8758:5: error: non-void function 'main' should return a value
      [-Wreturn-type]
    return  ;
    ^
4 warnings and 2 errors generated.
carp: callCommand: clang -fPIC -lm -g /Users/veitheller/.carp/out/main.c -o "/Users/veitheller/.carp/out/Untitled"  -I/Users/veitheller/Documents/Code/Github/carp/Carp//core/   (exit 1): failed
```

Reply:
This was not fixed, was it? Should I take this on or do you want to, @jacereda?

Reply:
I think I fixed it in some other place (`cross` or `tcc` PRs).

Reply:
Ah cool, nevermind then! (And thank you)

Reply:
Nope, looks like it should be in a PC where I did some experiments compiling with Visual C. I think the change was just emitting via something like:

```
show (ord c) ++ "/*" ++ show c ++ "*/"
```

Reply:
Oh cool!

Maybe we should disallow it in the parser, though, using something like `isSpace` from `Data.Char` (that should work on Unicode whitespace)? Whitespace variables don’t quite feel right to me.

Reply:
Ah, I thought the problem was related to values. No, I haven't worked on that, but sounds like `isSpace` could do the trick.